include/zephyr/drivers/adc/adc_current_sense_shunt.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_ADC_CURRENT_SENSE_SHUNT_H_
+#define ZEPHYR_INCLUDE_DRIVERS_ADC_CURRENT_SENSE_SHUNT_H_
+
+#include <zephyr/drivers/adc.h>
+
+struct current_sense_shunt_dt_spec {
+	const struct adc_dt_spec port;
+	int shunt_micro_ohms;
+	int shunt_gain;
+};
+
+/**
+ * @brief Get current sensor information from devicetree.
+ *
+ * This returns a static initializer for a @p current_sense_shunt_dt_spec structure
+ * given a devicetree node.
+ *
+ * @param node_id Devicetree node identifier.
+ *
+ * @return Static initializer for an current_sense_shunt_dt_spec structure.
+ */
+#define CURRENT_SENSE_SHUNT_DT_SPEC_GET(node_id)                                    \
+	{                                                                                          \
+		.port = ADC_DT_SPEC_GET(node_id),                                                  \
+		.shunt_micro_ohms = DT_PROP(node_id, shunt_resistor_micro_ohms),                   \
+		.shunt_gain = DT_PROP(node_id, gain),                                              \
+	}
+
+/**
+ * @brief Calculates the actual voltage from the measured voltage
+ *
+ * @param[in] spec current sensor specification from Devicetree.
+ * @param[in,out] v_to_c Pointer to the measured voltage in millivolts on input, and the
+ * corresponding scaled current value in milliamps on output.
```

Comment:
Should be v_to_i, conventional symbol for current is "I" (https://en.wikipedia.org/wiki/International_System_of_Units, https://en.wikipedia.org/wiki/SI_base_unit), C would be Coulomb.